File: swift_reference/Views/DayOfWeekPicker.swift
import SwiftUI

struct DayOfWeekPicker: View {
    @Binding var selectedDays: Set<Int>
    var isAbstain: Bool = false

    private let days: [(id: Int, label: String)] = [
        (1, "S"), (2, "M"), (3, "T"), (4, "W"), (5, "T"), (6, "F"), (7, "S")
    ]

    var body: some View {
        VStack(alignment: .leading, spacing: 10) {
            Text(isAbstain ? "Which days are you committing to this?" : "Which days will you do this?")
                .font(.caption.weight(.medium))
                .foregroundStyle(TributeColor.softGold.opacity(0.6))

            HStack(spacing: 8) {
                ForEach(days, id: \.id) { day in
                    let isSelected = selectedDays.contains(day.id)
                    Button {
                        withAnimation(.snappy(duration: 0.2)) {
                            if isSelected {
                                if selectedDays.count > 1 {
                                    selectedDays.remove(day.id)
                                }
                            } else {
                                selectedDays.insert(day.id)
                            }
                        }
                    } label: {
                        Text(day.label)
                            .font(.system(.subheadline, weight: .semibold))
                            .foregroundStyle(isSelected ? TributeColor.charcoal : TributeColor.softGold.opacity(0.5))
                            .frame(width: 38, height: 38)
                            .background(isSelected ? TributeColor.golden : TributeColor.surfaceOverlay)
                            .clipShape(Circle())
                    }
                    .buttonStyle(.plain)
                    .sensoryFeedback(.selection, trigger: isSelected)
                }
            }

            if selectedDays.count < 7 {
                Text(daysDescription)
                    .font(.caption2)
                    .foregroundStyle(TributeColor.softGold.opacity(0.4))
            }
        }
    }

    private var daysDescription: String {
        let names = [1: "Sun", 2: "Mon", 3: "Tue", 4: "Wed", 5: "Thu", 6: "Fri", 7: "Sat"]
        let sorted = selectedDays.sorted()
        let dayNames = sorted.compactMap { names[$0] }
        return "\(dayNames.joined(separator: ", ")) · \(selectedDays.count) days/week"
    }
}

File: lib/presentation/views/shared/day_of_week_picker.dart
import 'package:flutter/material.dart';
import '../../theme/app_theme.dart';

class DayOfWeekPicker extends StatelessWidget {
  final Set<int> selected;
  final ValueChanged<Set<int>> onChanged;
  final bool isAbstain;

  const DayOfWeekPicker({
    super.key,
    required this.selected,
    required this.onChanged,
    this.isAbstain = false,
  });

  static const _days = [
    (id: 1, label: 'S'),
    (id: 2, label: 'M'),
    (id: 3, label: 'T'),
    (id: 4, label: 'W'),
    (id: 5, label: 'T'),
    (id: 6, label: 'F'),
    (id: 7, label: 'S'),
  ];

  static const _names = {1: 'Sun', 2: 'Mon', 3: 'Tue', 4: 'Wed', 5: 'Thu', 6: 'Fri', 7: 'Sat'};

  String get _description {
    final sorted = selected.toList()..sort();
    final names = sorted.map((d) => _names[d] ?? '').where((s) => s.isNotEmpty);
    return '${names.join(', ')} · ${selected.length} days/week';
  }

  @override
  Widget build(BuildContext context) {
    return Column(
      crossAxisAlignment: CrossAxisAlignment.start,
      children: [
        Text(
          isAbstain ? 'Which days are you committing to this?' : 'Which days will you do this?',
          style: TextStyle(
            color: MyWalkColor.softGold.withValues(alpha: 0.6),
            fontSize: 12,
            fontWeight: FontWeight.w500,
          ),
        ),
        const SizedBox(height: 10),
        Row(
          mainAxisAlignment: MainAxisAlignment.spaceBetween,
          children: _days.map((day) {
            final isSelected = selected.contains(day.id);
            return GestureDetector(
              onTap: () {
                final updated = Set<int>.from(selected);
                if (isSelected) {
                  if (updated.length > 1) updated.remove(day.id);
                } else {
                  updated.add(day.id);
                }
                onChanged(updated);
              },
              child: AnimatedContainer(
                duration: const Duration(milliseconds: 150),
                width: 38,
                height: 38,
                decoration: BoxDecoration(
                  shape: BoxShape.circle,
                  color: isSelected ? MyWalkColor.golden : MyWalkColor.surfaceOverlay,
                ),
                alignment: Alignment.center,
                child: Text(
                  day.label,
                  style: TextStyle(
                    color: isSelected ? MyWalkColor.charcoal : MyWalkColor.softGold.withValues(alpha: 0.5),
                    fontWeight: FontWeight.w600,
                    fontSize: 14,
                  ),
                ),
              ),
            );
          }).toList(),
        ),
        if (selected.length < 7) ...[
          const SizedBox(height: 6),
          Text(
            _description,
            style: TextStyle(
              color: MyWalkColor.softGold.withValues(alpha: 0.4),
              fontSize: 10,
            ),
          ),
        ],
      ],
    );
  }
}
